sentinel/src/transitions/accept_patient_reports.js

```diff
@@ -114,6 +114,45 @@ const addRegistrationToDoc = (doc, registrations) => {
   }
 };
 
+const findValidRegistration = (doc, registrations) => {
+    const visitReportedDate = doc.reported_date;
+
+    for (var i = 0; i < registrations.length; i++) {
+      var registration = registrations[i];
+      if (registration.scheduled_tasks) {
+        for (var j = 0; j < registration.scheduled_tasks.length; j++) {
+          var task = registration.scheduled_tasks[j];
+          if (['delivered', 'sent'].includes(task.state)) {
+            var nextTask = registration.scheduled_tasks[j + 1] || { due: moment(visitReportedDate).add(1, 'd') };
+            if (nextTask && moment(nextTask.due) > moment(visitReportedDate)) {
+              // We loop through tasks. Once we find one that has either the status "delivered" or "sent" with
+              // a future task with a due date that is after the visit reported date then we have found the task 
+              // linked to the visit. We always link if this is the last scheduled task delivered or sent.
+              registration.scheduled_tasks[j].report_uuid = doc._id;
+              return registration;
+            }
+          }          
+        }
+      }
+    } 
+
+    return null;
```

Comment:
You don't need this: in Javascript, if there is a function that doesn't return anything its return value will be `undefined`, which is [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy), and so will be equivalent to `null`.